docs/api/js/types.md

```diff
@@ -1,226 +1,303 @@
-    provenAt?: Date;
-    executeTxHash?: string;
-    executedAt?: Date;
-    l1GasPrice: number;
-    l2FairGasPrice: number;
-}
-
-export interface BlockDetails {
-    number: number;
-    timestamp: number;
-    l1BatchNumber: number;
-    l1TxCount: number;
-    l2TxCount: number;
-    rootHash?: string;
-    status: string;
-    commitTxHash?: string;
-    committedAt?: Date;
-    proveTxHash?: string;
-    provenAt?: Date;
-    executeTxHash?: string;
-    executedAt?: Date;
-}
-
-export interface TransactionDetails {
-    isL1Originated: boolean;
-    status: string;
-    fee: BigNumberish;
-    initiatorAddress: Address;
-    receivedAt: Date;
-    ethCommitTxHash?: string;
-    ethProveTxHash?: string;
-    ethExecuteTxHash?: string;
-}
-```
+# Types and interfaces
+
+## `AccountAbstractionVersion`
+
+Enumerated list of account abstraction versions.
+
+- None = `0` (used for contracts that are not accounts)
+- Version1 = `1`
+
+## `AccountNonceOrdering`
+
+Enumerated list of account nonce ordering formats.
+
+- Sequential = `0`
+- Arbitrary = `1`
+
+## `Address`
+
+0x-prefixed, hex-encoded, Ethereum account address as string.
+
+## `ApprovalBasedPaymasterInput`
+
+Interface representation of approval based paymaster input containing various fields including an `ApprovalBased` type.
+
+- `type`: `'ApprovalBased'`;
+- `token`: `Address`;
+- `minimalAllowance`: `BigNumber`;
+- `innerInput`: `BytesLike`;
+
+## `BalancesMap`
+
+Type defining a map object containing accounts and their balances.
+
+- `{ [key: string]`: `BigNumber }`
+
+## `BatchDetails`
+
+Interface representation of batch information containing various optional and mandatory fields.
+
+- `number`: `number`;
+- `timestamp`: `number`;
+- `l1TxCount`: `number`;
+- `l2TxCount`: `number`;
+- `rootHash?`: `string`;
+- `status`: `string`;
+- `commitTxHash?`: `string`;
+- `committedAt?`: `Date`;
+- `proveTxHash?`: `string`;
+- `provenAt?`: `Date`;
+- `executeTxHash?`: `string`;
+- `executedAt?`: `Date`;
+- `l1GasPrice`: `number`;
+- `l2FairGasPrice`: `number`;
+
+## `Block`
+
+Interface representation of a block that extends the Ethers `providers.Block` definition with additional fields.
+
+- `l1BatchNumber`: `number`;
+- `l1BatchTimestamp`: `number`;
+
+## `BlockDetails`
+
+Interface representation of block information containing various optional and mandatory fields.
+
+- `number`: `number`;
+- `timestamp`: `number`;
+- `l1BatchNumber`: `number`;
+- `l1TxCount`: `number`;
+- `l2TxCount`: `number`;
+- `rootHash?`: `string`;
+- `status`: `string`;
+- `commitTxHash?`: `string`;
+- `committedAt?`: `Date`;
+- `proveTxHash?`: `string`;
+- `provenAt?`: `Date`;
+- `executeTxHash?`: `string`;
+- `executedAt?`: `Date`;
+
+## `BlockTag`
+
+Pipe-delimited list of block labels that includes block number in denary and hex plus block statuses.
+
+- `number`
+- `string` // hex representation of block number
+- `committed`
+- `finalized`
+- `latest`
+- `earliest`
+- `pending`
```

Comment:
they should be fin into quotes like 
`'latest'`

Reply:
DONE in new branch.